### PR TITLE
Updating Webhook and User Import Schema for links and codes

### PR DESCRIPTION
**Pull request by DerekSiemens**
## Description of the change
Updating webhook schema and examples with the new changes from Vanity Links and Codes

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation or Development tools (readme, specs, tests, code formatting)

## Links

 - Jira issue number: ji2783
 - Process.st launch checklist: https://app.process.st/checklists/Vanity-Links-and-Codes-m2vL63FIM9e3QB_sN5JIAQ/tasks/onlBImyM_Q-rfZjJGw9HwA

### Development

- [x] [Prettier](https://www.npmjs.com/package/prettier) was run
- [ ] The behaviour changes in the pull request are covered by specs
- [ ] All tests related to the changed code pass in development

### Paperwork

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has a Jira number
- [x] This pull request has a Process.st launch checklist
- [ ] "Ready for review" label attached to the PR and reviewers pinged on Slack

### Code review 

- [x] Changes have been reviewed by at least one other engineer
- [x] Security impacts of this change have been considered

### CODE REVIEW

File: src/json/webhooks/ReferralClassicDisabled.schema.json [hunk 1]
```diff
@@ -45,6 +55,13 @@
       "description": "The Id of the program this code was created for"
     }
   },
-  "required": ["id", "referrer", "referred", "dateReferralStarted"],
+  "required": [
+    "id",
+    "referrer",
+    "referred",
+    "referralCodeUsed",
+    "shareLinkUsed",
```
Comment (johanventer) [inline src/json/webhooks/ReferralClassicDisabled.schema.json:63]: @DerekSiemens I'm not sure `shareLinkUsed` should be required here - it's possible to create referrals without a share link, i.e. from the portal. Can you please check with @slisaasquatch on this one?

Comment (DerekSiemens) [reply src/json/webhooks/ReferralClassicDisabled.schema.json:63]: Good point, Ill check with him

File: src/json/webhooks/ReferralClassicDisabled.schema.json [hunk 2]
```diff
@@ -45,6 +55,11 @@
       "description": "The Id of the program this code was created for"
     }
   },
-  "required": ["id", "referrer", "referred", "dateReferralStarted"],
+  "required": [
+    "id",
+    "referrer",
+    "referred",
```
Comment (johanventer) [inline src/json/webhooks/ReferralClassicDisabled.schema.json:61]: I think we discussed `referred` not being a required field here? Please check.

Comment (DerekSiemens) [reply src/json/webhooks/ReferralClassicDisabled.schema.json:61]: I thought that was only for classic, but ill double check with Scott

Comment (DerekSiemens) [reply src/json/webhooks/ReferralClassicDisabled.schema.json:61]: Just checked, only not required for classic

File: src/json/webhooks/UserCreatedClassicDisabled.schema.json [hunk 3]
```diff
@@ -94,6 +95,7 @@
       }
     },
     "programShareLinks": {
+      "description": "The users primary share links for the primary code of each program",
```
Comment (loganvolkers) [inline src/json/webhooks/UserCreatedClassicDisabled.schema.json:98]: @DerekSiemens I don't think is valid schema? I think if you use `$ref` like this you have to use `allOf` too?

**Comment by DerekSiemens** [comment]
Apologies for the late update on that last commit

**Pull request merged**